### DISCUSSION TITLE
../../src/libstrongswan/.libs/libstrongswan.so: undefined reference to `_dl_pagesize' Strongswan-5.9.13 is failing to compile using arm-none-linux-gnueabi-gcc toolchain

### DISCUSSION THREAD

**Discussion by mdmuazzamhusain** [strongswan/strongswan#2107] [General]
Hi All,
I am using below commands to compile package, but somehow it's failing.
**sudo ./configure --prefix=/opt/strongswan --host=arm-none-linux-gnueabi CC=arm-none-linux-gnueabi-gcc --libexecdir=/opt/strongswan/libexec --sysconfdir=/opt/strongswan/etc --enable-eap-sim --enable-eap-sim-file --enable-eap-identity --enable-openssl --libdir=/opt/strongswan --enable-ctr --enable-pkcs11 --enable-xauth-noauth --enable-xauth-pam --enable-eap-mschapv2 --enable-eap-gtc --enable-eap-md5  -enable-eap-aka-3gpp2 --enable-blowfish --enable-md4 --enable-dnscert --enable-ipseckey --enable-attr   CFLAGS="-I/tmp/install/include/ -march=armv7-a -I/tmp/install/eNB_include/include/" LIBS='-L/tmp/install/lib -lcrypto -pthread'**
followed by **make** command.

*** Warning: Linking the executable xfrmi against the loadable module
*** libstrongswan-kernel-netlink.so is not portable!
libtool: link: arm-none-linux-gnueabi-gcc -Wno-format -Wno-format-security -Wno-implicit-fallthrough -Wno-missing-field-initializers -Wno-pointer-sign -Wno-sign-compare -Wno-type-limits -Wno-unused-parameter -I/tmp/install/include/ -march=armv7-a -I/tmp/install/eNB_include/include/ -include /home/build/mhusain/strongswan-5.9.13/config.h -o .libs/xfrmi xfrmi.o  -L/opt/strongswan ../../src/libcharon/plugins/kernel_netlink/.libs/libstrongswan-kernel-netlink.so ../../src/libstrongswan/.libs/libstrongswan.so ../../src/libcharon/.libs/libcharon.so -L/tmp/install/lib -lcrypto -lpthread -pthread -Wl,-rpath -Wl,/opt/strongswan/ipsec/plugins -Wl,-rpath -Wl,/opt/strongswan/ipsec
../../src/libstrongswan/.libs/libstrongswan.so: undefined reference to `_dl_pagesize'
../../src/libstrongswan/.libs/libstrongswan.so: undefined reference to `_dl_init_static_tls'
../../src/libstrongswan/.libs/libstrongswan.so: undefined reference to `__libc_setup_tls'
../../src/libstrongswan/.libs/libstrongswan.so: undefined reference to `_dl_stack_flags'
../../src/libstrongswan/.libs/libstrongswan.so: undefined reference to `_dl_osversion'
../../src/libstrongswan/.libs/libstrongswan.so: undefined reference to `_dl_wait_lookup_done'
collect2: ld returned 1 exit status
make[3]: *** [xfrmi] Error 1
make[3]: Leaving directory `/home/build/mhusain/strongswan-5.9.13/src/xfrmi'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/home/build/mhusain/strongswan-5.9.13/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/build/mhusain/strongswan-5.9.13'
make: *** [all] Error 2

**Comment by tobiasbrunner** [comment]
Can't really help you with this. Sounds like a cross-compile/toolchain issue (maybe specifying `LIBS` isn't a good idea).

**Comment by mdmuazzamhusain** [comment]
Hi Tobias,
              Thanks for your quick response.
              Actually using same toolchain, previously 5.3.2 pkg was compiled, same also not getting compiled.  For below error if I use -lpthread in configure script option then at least it goes ahead. But again stuck with same error what I have posted on community.

              Error log:
              make[5]: Entering directory `/home/build/mhusain/strongswan-5.3.2/src/starter'
/bin/sh ../../libtool  --tag=CC   --mode=link arm-none-linux-gnueabi-gcc  -I/tmp/install/include -march=armv7-a -I/tmp/install/eNB_include/include/ -include /home/build/mhusain/strongswan-5.3.2/config.h  -L/tmp/install/lib -L/opt/strongswan -o starter starter.o args.o confread.o keywords.o cmp.o invokecharon.o starterstroke.o netkey.o klips.o ../../src/libstrongswan/libstrongswan.la ../../src/libhydra/libhydra.la libstarter.la   -lcrypto
libtool: link: arm-none-linux-gnueabi-gcc -I/tmp/install/include -march=armv7-a -I/tmp/install/eNB_include/include/ -include /home/build/mhusain/strongswan-5.3.2/config.h -o .libs/starter starter.o args.o confread.o keywords.o cmp.o invokecharon.o starterstroke.o netkey.o klips.o  -L/tmp/install/lib -L/opt/strongswan ../../src/libstrongswan/.libs/libstrongswan.so ../../src/libhydra/.libs/libhydra.so ./.libs/libstarter.a -lcrypto -Wl,-rpath -Wl,/opt/strongswan/ipsec
/home/build/CodeSourcery/Sourcery_CodeBench_Lite_for_ARM_GNU_Linux/bin/../lib/gcc/arm-none-linux-gnueabi/4.6.3/../../../../arm-none-linux-gnueabi/bin/ld: starter.o: undefined reference to symbol 'pthread_sigmask@@GLIBC_2.4'
/home/build/CodeSourcery/Sourcery_CodeBench_Lite_for_ARM_GNU_Linux/bin/../lib/gcc/arm-none-linux-gnueabi/4.6.3/../../../../arm-none-linux-gnueabi/bin/ld: note: 'pthread_sigmask@@GLIBC_2.4' is defined in DSO /home/build/CodeSourcery/Sourcery_CodeBench_Lite_for_ARM_GNU_Linux/bin/../arm-none-linux-gnueabi/libc/lib/libpthread.so.0 so try adding it to the linker command line
/home/build/CodeSourcery/Sourcery_CodeBench_Lite_for_ARM_GNU_Linux/bin/../arm-none-linux-gnueabi/libc/lib/libpthread.so.0: could not read symbols: Invalid operation
collect2: ld returned 1 exit status


Thanks,
Muazzam
